PROJECT GOALS:
1. Browse the dataset, inspect its shape, features, data types
2. Formulate valid questions to analyze through the available data

In [2]:
import pandas as pd
import numpy as np

In [16]:
data_file = pd.read_csv('profiles.csv')
print('''###############################################
DataFrame info:''')
print(data_file.info())
print('''###############################################
DataFrame sample:''')
#data_file.head()
features_of_interest = ['body_type', 'job','diet', 'drinks', 'drugs', 'education', 'orientation','sex', 'smokes', 'status', 'religion', 'ethnicity']

#removing unwanted features
data_file = data_file.drop(columns=['essay0','essay1','essay2','essay3','essay4','essay5','essay6','essay7','essay8','essay9','last_online'])
for i in data_file.columns:
    uniques= data_file[i].nunique()
    print(f'{i} uniques: {uniques}')
# data_file.head()
    


###############################################
DataFrame info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 59946 entries, 0 to 59945
Data columns (total 31 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   age          59946 non-null  int64  
 1   body_type    54650 non-null  object 
 2   diet         35551 non-null  object 
 3   drinks       56961 non-null  object 
 4   drugs        45866 non-null  object 
 5   education    53318 non-null  object 
 6   essay0       54458 non-null  object 
 7   essay1       52374 non-null  object 
 8   essay2       50308 non-null  object 
 9   essay3       48470 non-null  object 
 10  essay4       49409 non-null  object 
 11  essay5       49096 non-null  object 
 12  essay6       46175 non-null  object 
 13  essay7       47495 non-null  object 
 14  essay8       40721 non-null  object 
 15  essay9       47343 non-null  object 
 16  ethnicity    54266 non-null  object 
 17  height       59943 non-n

Possible predictions:
1. Predict the zodiac sign
2. Predict sex
3. Predict religion
4. Predict drugs
5. Predict your ideal partner's age based on other criteria
6. Predict your ideal partner's religion based on other criteria

Data Cleanup:
age          59946 non-null  int64  - check data distribution
body_type    54650 non-null  object - create categories, check distribution
diet         35551 non-null  object - bin data, check distribution
drinks       56961 non-null  object - create categories, check distribution
drugs        45866 non-null  object - create categories, check distribution
education    53318 non-null  object - bin data, check distribution
ethnicity    54266 non-null  object - ? too many combinations
height       59943 non-null  float64 - check data distribution
income       59946 non-null  int64  - check data distribution
job          51748 non-null  object - check data distribution
last_online  59946 non-null  object - disregard
location     59946 non-null  object - bin data, check distribution
offspring    24385 non-null  object - check data distribution, probably disregard due to high nan count
orientation  59946 non-null  object  - check data distribution
pets         40025 non-null  object  - check data distribution
religion     39720 non-null  object  - check data distribution
sex          59946 non-null  object  - check data distribution
sign         48890 non-null  object  - check data distribution
smokes       54434 non-null  object - create categories, check distribution
speaks       59896 non-null  object - bin data, check distribution
status       59946 non-null  object - check data distribution